### PR TITLE
New credit card form

### PR DESCRIPTION
**Pull request by nuxzero**
### Objective

**AppCompat theme**

The objective of this PR is to update the CreditCardActivity to support the AppCompat theme from the support library. 

**Update UI components**

The last thing is to update the UI components used on the credit card form to the new design.

**Drop Card IO**

Noticed the CardIO had no longer updated for a year, and the repo status had changed to Archived. Also, on the Omise iOS already dropped it from the project. So, I decided to drop the CardIO from the project.

### Description of changes

**AppCompat theme**

On the CreditCardActivity had changed the parent class to AppCompatActivity to support AppCompat theme. And using the `Appcompat.Light` as the default theme. 

**Update UI components**

For the UI components, contains creating, updating, and deprecating new UI components for the credit card form.

- **Creating** In this PR, I created `OmiseEditText` as based class  that contain `validate()` function to check input validation. Also, created `CardNameEditText`, `ExpiryDateEditText`, and `SecurityCodeEditText` with that contain validation function and formating input.

- **Updating** Updated `CreditCardEditText` to show the brand logo when card number match with the brand.

- **Deprecating** The expiry date spinner marked as deprecated on this PR. And, introduce the `ExpiryDateEditText` as the new expiry date input.

**Drop Card IO**

Deleted the scan card menu from the CreditCardActivity. Also, dropped Card IO from the `build.gradle` file and deleted all code that involved to Card IO.

### QA

The sample app had no change on the starting `CreditCardActivity`. 
The new `CreditCardActivity` should appear with the new look to more Material Design style.

new-credit-card-form

> ### NOTE
>
> On this PR doesn't include the theming and some of new come up things on the new design. It will continue with another task.

### CODE REVIEW

File: sdk/src/main/java/co/omise/android/ui/CreditCardActivity.kt [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+            it.setOnAfterTextChangeListener(::updateSubmitButton)
+        }
+    }
+
+    override fun onBackPressed() {
+        setResult(RESULT_CANCELED)
+        super.onBackPressed()
+    }
+
+    private inner class ActivityRequestListener : RequestListener<Token> {
```
Comment (Namiii) [inline sdk/src/main/java/co/omise/android/ui/CreditCardActivity.kt]: Perhaps we can use a better name here? `CreditCardRequestListener` or just `listener`?

File: sdk/src/main/java/co/omise/android/ui/CreditCardActivity.kt [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+            throw IllegalAccessException("Can not found ${::EXTRA_PKEY.name}.")
+        }
+
+        setContentView(R.layout.activity_credit_card)
+        setTitle(R.string.default_form_title)
+
+        submitButton.setOnClickListener(::submit)
+
+        editTexts.forEach {
+            it.setOnFocusChangeListener { _, hasFocus ->
```
Comment (Namiii) [inline sdk/src/main/java/co/omise/android/ui/CreditCardActivity.kt]: I really like how this works here, kudos 👍

Comment (igroomgrim) [reply sdk/src/main/java/co/omise/android/ui/CreditCardActivity.kt]: 🍭

File: app/src/main/res/values/styles.xml [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+        <item name="editTextStyle">@style/AppEditText</item>
+    </style>
+
+    <style name="AppActionBar" parent="Widget.AppCompat.Toolbar">
+        <item name="android:background">@android:color/white</item>
+        <item name="titleTextColor">@android:color/black</item>
+    </style>
+
+    <style name="AppEditText" parent="Widget.AppCompat.EditText">
+        <item name="android:textSize">14sp</item>
```
[the remark this answers is not in the document]
Comment (nuxzero) [reply app/src/main/res/values/styles.xml]: I think it fine for sample app. BTW, I'll extract to `dimen`.

File: sdk/src/main/java/co/omise/android/ui/OmiseEditText.kt [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+    var errorMessage: String? = null
+        set(value) {
+            field = value
+            invalidate()
+        }
+
+    init {
+        errorText = TextPaint().apply {
+            color = Color.RED
+            textSize = 14f
```
[the remark this answers is not in the document]
Comment (nuxzero) [reply sdk/src/main/java/co/omise/android/ui/OmiseEditText.kt]: As we discussed, I'll delete the error message from `OmiseEditText` and use normal `TextView` instead.

So, I'll keep it likes this and delete it in the next task.

File: sdk/src/main/java/co/omise/android/ui/OmiseEditText.kt [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+            textAlign = Paint.Align.LEFT
+        }
+    }
+
+    override fun onDraw(canvas: Canvas?) {
+        super.onDraw(canvas)
+
+        val xPos = paddingLeft.toFloat()
+        val yPos = height - (errorText?.fontMetrics?.bottom ?: 0f)
+        canvas?.drawText(
```
[the remark this answers is not in the document]
Comment (nuxzero) [reply sdk/src/main/java/co/omise/android/ui/OmiseEditText.kt]: Same above comment.

File: sdk/src/main/java/co/omise/android/ui/CreditCardActivity.kt (hunk 1 above)
Comment (nuxzero) [reply sdk/src/main/java/co/omise/android/ui/CreditCardActivity.kt]: Yeah, I'll change this.

File: sdk/src/main/java/co/omise/android/ui/OmiseEditText.kt [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+            try {
+                validate()
+                true
+            } catch (e: InputValidationException) {
+                false
+            }
+
+    @Throws(InputValidationException::class)
+    open fun validate() {
+        val value = text.toString().trim { it <= ' ' }
```
[the remark this answers is not in the document]
Comment (nuxzero) [reply sdk/src/main/java/co/omise/android/ui/OmiseEditText.kt]: Yeah, it just check empty space. I'll change it to normal `trim()`.

**Comment by nuxzero** [comment]
@Namiii I made some code changed from your comments. Could you help me review again please? 🙏

File: app/src/main/res/values/styles.xml (hunk 3 above)
[the remark this answers is not in the document]
Comment (Namiii) [reply app/src/main/res/values/styles.xml]: Ohhh I didn't realize it was the sample app 🤦‍♂

File: sdk/src/main/java/co/omise/android/ui/OmiseEditText.kt (hunk 4 above)
[the remark this answers is not in the document]
Comment (Namiii) [reply sdk/src/main/java/co/omise/android/ui/OmiseEditText.kt]: 👍

**Pull request merged**